From fab81b199ca0ae69a99bbb20f951d8a3b997c679 Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Thu, 8 Jul 2021 02:02:50 +0300
Subject: [PATCH 3/3] arm64: dts: renesas: r8a77980: Remove redundant isp0 and
 isp1 nodes

This removes isp[01] nodes, and adds "renesas,isp-r8a77980"
to the tisp[01] compatible property. Now tisp[01] nodes should
be used instead of isp0. This prevents resource conflict when
both rcar-isp and uio_pdrv_genirq drivers are used.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 arch/arm64/boot/dts/renesas/r8a77980.dtsi | 28 ++++------------------------
 1 file changed, 4 insertions(+), 24 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a77980.dtsi b/arch/arm64/boot/dts/renesas/r8a77980.dtsi
index 9f3bc6c..bb954a9 100644
--- a/arch/arm64/boot/dts/renesas/r8a77980.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77980.dtsi
@@ -24,8 +24,6 @@
 		i2c3 = &i2c3;
 		i2c4 = &i2c4;
 		i2c5 = &i2c5;
-		isp0 = &isp0;
-		isp1 = &isp1;
 		tisp0 = &tisp0;
 		tisp1 = &tisp1;
 		cisp0 = &cisp0;
@@ -1776,28 +1774,9 @@
 			reg = <0 0xf12a0000 0 0x1000>;
 		};
 
-		isp0: isp@fec00000 {
-			compatible = "renesas,isp-r8a77980";
-			reg = <0 0xfec00000 0 0x20000>,
-			      <0 0xfed00000 0 0x10000>;
-			interrupts = <GIC_SPI 26 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 25 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 817>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-		};
-
-		isp1: isp@fee00000 {
-			compatible = "renesas,isp-r8a77980";
-			reg = <0 0xfee00000 0 0x20000>,
-			      <0 0xfed20000 0 0x10000>;
-			interrupts = <GIC_SPI 109 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 27 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 814>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-		};
-
 		tisp0: tisp_00 {
-			compatible = "generic-uio";
+			compatible = "renesas,isp-r8a77980",
+				     "generic-uio";
 			reg = <0 0xfed00000 0 0x10000>;
 			interrupts = <GIC_SPI 25 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&cpg CPG_MOD 817>;
@@ -1816,7 +1795,8 @@
 		};
 
 		tisp1: tisp_01 {
-			compatible = "generic-uio";
+			compatible = "renesas,isp-r8a77980",
+				     "generic-uio";
 			reg = <0 0xfed20000 0 0x10000>;
 			interrupts = <GIC_SPI 27 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&cpg CPG_MOD 814>;
-- 
2.7.4

